src/storage/btree.c

```diff
@@ -4280,13 +4302,171 @@ btree_read_record (THREAD_ENTRY * thread_p, BTID_INT * btid, PAGE_PTR pgptr, REC
 	}
       else if (n_prefix < 0)
 	{
+	  assert (0);
+	  return n_prefix;
+	}
+    }
+
+  return NO_ERROR;
+}
+
+
+#if !defined(NDEBUG) && defined(IMPROVE_RANGE_SCAN_IN_BTREE)
+static void
+check_validate (BTREE_SCAN * bts)
+{
+  if (bts->is_cur_key_compressed)
+    {
+      assert (bts->common_prefix_size > 0);
+      assert (VPID_ISNULL (&bts->C_vpid) == false);
+      assert (VPID_EQ (&(bts->C_vpid), &bts->cur_common_prefix_page_vpid) == true);
+      assert (bts->C_page != NULL);
+#if defined(CHECK_VERIFY_COMMON_PREFIX_PAGE_INFO)
+      assert (VPID_ISNULL (&bts->cur_common_prefix_page_vpid) == false);
+      assert (LSA_EQ (&bts->cur_common_prefix_page_lsa, pgbuf_get_lsa (bts->C_page)));
+#endif
+    }
+}
+#else
+#define check_validate(bts) ;
+#endif
+
+#if defined(IMPROVE_RANGE_SCAN_IN_BTREE)
+static int
+btree_read_record_in_leafpage (THREAD_ENTRY * thread_p, PAGE_PTR pgptr, int copy_key, BTREE_SCAN * bts)
```

Comment:
The meaning of the names of the functions `btree_read_record` and `btree_read_record_without_decompression` is clear.
However, the `btree_read_record_in_leafpage` function does not.

How about adding the code from the `btree_read_record_in_leafpage` function to the `btree_read_record_without_decompression` function?

Reply:
Currently, btree_read_record_in_leafpage() is called only from btree_range_scan_read_record(). So you could put your code directly in btree_range_scan_read_record().

However, since it needs to be called in the [CBRD-25159] issue that will be added later, we will keep a separate name.
btree_read_record() reads keys from all nodes of B-Tree, but this function is named to mean that it only reads keys from leaf nodes.

Please recommend a good name.

Reply:
The btree_read_record function reads records without distinguishing between leaf nodes and non-leaf nodes.
However, it is ambiguous that there is an additional btree_read_record_in_leafpage function to read records only from leaf nodes.
There is no ambiguity if the btree_read_record function calls the btree_read_record_in_leafpage function.

The purpose is to express my opinion, so if you read it, please resolve it.

Reply:
This will be covered again in [CBRD-25159].
I will review it again then.
Thank you